enhancements/network/allow-mtu-changes.md

```diff
@@ -0,0 +1,319 @@
+  from host network to the cluster pod network.
+
+On startup, CNI should reset the MTU value on interfaces and routes to the
+current setting of `mtu` and `routable-mtu` respectively even if such routes
+already exist. If `routable-mtu` is unset, the CNI should unset the MTU on the
+routes.
+
+The CNI should annotate the `mtu` and `routable-mtu` values used on each node
+that will allow to track the mtu-migration procedure.
+
+### MCO: ability to configure host MTU migration
+
+A configuration file will be deployed through MCO with the MTU parameters that
+need to be set during MTU migration:
+* `TARGET_MTU` should be persistently set in the network configuration of the
+  appropriate host interfaces. This setting should persist even if `TARGET_MTU`
+  is unset in a subsequent reboot.
+* `MTU` should be dynamically set in the appropriate host interfaces. If `MTU`
+  is not set, use the highest of `TARGET_MTU` or current interface MTU.
+* `ROUTABLE_MTU` should be dynamically set for link scoped routes of the
+  appropriate host interfaces. If `ROUTABLE_MTU` is not set, use the lowest of
+  `TARGET_MTU`, current interface MTU and current route MTU.
+
+Host tooling that configures host networking for the CNI should source this
+configuration file and act accordingly. 
+
+Host tooling should write a file for NFD source hook to setup annotations for
+current MTU, minimum MTU, maximum MTU and routable MTU if any.
+
+### openshift-api: add MTU migration specific fields
+
+New `MTU.Network.From`, `MTU.Network.To` and `MTU.Machine.To` fields will be
+added in the existing `NetworkMigration` type which is used in CNO's cluster and
+operator network configurations and will contain the CNI and host MTU values to
+migrate from and to to respectively. 
+
+### MCO: automating MTU migration rolling reboots
+
+MCO monitors `Network.Status.Migration` from the cluster network configuration
+configuration. This `Network.Status.Migration` is already updated by CNO with
+the validated contents of `Network.Spec.Migration` from the operator network
+configuration. MCO should react to changes in `MTU.Machine.To` and render the
+appropriate MachineConfig containing the MTU configuration file with
+`TARGET_MTU` set to `MTU.Machine.To` if:
+* If `MTU.Machine.To` is set and `MTU.Network.To` is not set, for the case of
+  host MTU migration with no CNI MTU migration.
+* If `MTU.Machine.To` is set and, `MTU.Network.To` is set and different than
+  `Network.Status.ClusterNetworkMTU`, for the case of host MTU migration with
+  CNI MTU migration.
+
+This will result in the required rolling reboots as `TARGET_MTU` is set and then
+unset during the MTU migration procedure.
+
+### CNO: allowing MTU migration
+
+The CNO monitors changes on the operator network configuration. When it detects
+a change in `Network.Spec.Migration`:
+1. Check that `MTU.Network.From` and `MTU.Network.To` are not independently set
```

Comment:
why does this say "or different". Shouldn't what you are migrating From and migrating To be different?

Reply:
Fixed. Actually To and From can be equal to attempt to rollback a MTU migration that is in progress when something goes wrong.